src/main/java/quotify_app/usecases/landing/LandingInteractor.java

```diff
@@ -0,0 +1,93 @@
+package quotify_app.usecases.landing;
+
+import java.util.List;
+
+import quotify_app.entities.regionEntities.Address;
+import quotify_app.entities.regionEntities.Area;
+import quotify_app.entities.regionEntities.Property;
+
+/**
+ * The Landing Interactor.
+ */
+public class LandingInteractor implements LandingInputBoundary {
+    private final AreaDataAccessInterface areaDataAccessObject;
+    private final PropertyDataAccessInterface propertyDataAccessObject;
+    private final LandingOutputBoundary landingPresenter;
+
+    public LandingInteractor(AreaDataAccessInterface areaDataAccessInterface,
+                             PropertyDataAccessInterface propertyDataAccessInterface,
+                             LandingOutputBoundary landingOutputBoundary) {
+        this.areaDataAccessObject = areaDataAccessInterface;
+        this.propertyDataAccessObject = propertyDataAccessInterface;
+        this.landingPresenter = landingOutputBoundary;
+    }
+
+    /**
+     * Fetches, caches and calls presenter with a list of available areas for the user to select.
+     * @param geoIdV4 the geoId of the selected parent area.
+     * @param type the type of subarea to be fetched.
+     */
+    @Override
+    public void fetchAreas(String geoIdV4, String type) {
+        try {
+            final List<Area> areaList = areaDataAccessObject.getSubAreas(geoIdV4, type);
+            areaDataAccessObject.cacheAreas(areaList, type);
+            final AreaListOutputData outputData = new AreaListOutputData(areaList, type, false);
+            landingPresenter.prepareAreaListSuccessView(outputData);
+        }
+        catch (Exception e) {
```

Comment:
does checkstyle not complain about this? I have a feeling it might later on -- can try adding a more specific type of exception here

Reply:
Absolutely, I'll fix all instances of catching Exception e.